keel-api/src/main/kotlin/com/netflix/spinnaker/keel/rest/ResourceController.kt

```diff
@@ -53,11 +54,17 @@ class ResourceController(
     consumes = [APPLICATION_YAML_VALUE, APPLICATION_JSON_VALUE],
     produces = [APPLICATION_YAML_VALUE, APPLICATION_JSON_VALUE]
   )
-  fun create(@RequestBody resource: Resource<*>): Resource<*> {
+  fun create(@RequestBody submittedResource: SubmittedResource<*>): Resource<*> {
     // TODO: we need to take the resource type as well so we can actually parse and validate here
-    log.info("Creating: $resource")
-    publisher.publishEvent(ResourceEvent(CREATE, resource))
-    return resource
+    // if you're creating a resource you don't need to pass the name
+    log.info("Creating: $submittedResource")
+    val resource = Resource(
+      metadata = ResourceMetadata(name = ResourceName("undefined")),
+      kind = submittedResource.kind,
+      apiVersion = submittedResource.apiVersion,
+      spec = submittedResource.spec
+    )
```

Comment:
Probably. I'll do that.

Reply:
Discussed offline: that actually makes things more complicated because it essentially makes `metadata` nullable. Leaving as is for now and we can refine later if needed.